lib/cstruct_cap.mli

```diff
@@ -150,21 +146,44 @@ val shift : 'a rd t -> int -> 'a rd t
 
     @raise Invalid_argument if the offset exceeds [t] length. *)
 
-val to_string : ?off:int -> ?len:int -> 'a rd t -> string
-(** [to_string ~off ~len t] is the string representation of the segment of [t]
-    starting at [off] (default is [0]) of size [len] (default is [length t]).
-    [t] needs at least read-capability {!rd}.
+val split : ?start:int -> 'a t -> int -> 'a t * 'a t
+(** [split ~start t len] is a tuple containing {!t}s extracted from [t] at
+    offset [start] (default is [0]) of length [len] as first element, and the
+    rest of [t] as second element. The function {!sub} is used, thus the
+    underlying buffer is shared.
 
-    @raise Invalid_argument if [off] and [len] does not designate a valid
-    segment of [t]. *)
+    @raise Invalid_argument if [start] exceeds the length of [t],
+    or if there is a bounds violation of [t] via [len + start]. *)
 
-val to_bytes : ?off:int -> ?len:int -> 'a rd t -> bytes
-(** [to_bytes ~off ~len t] is the bytes representation of the segment of [t]
-    starting at [off] (default is [0]) of size [len] (default is [length t]).
-    [t] needs at least read-capability {!rd}.
+(** {2 Fresh construction from existing t} *)
 
-    @raise Invalid_argument if [off] and [len] do not designate a valid
-    segment of [t]. *)
+val append : 'a rd t -> 'b rd t -> rdwr t
+(** [append a b] create a fresh {!t} which is the concatenation of [a] and [b].
+    [a] and [b] need at least read-capability {!rd}.
+    The return [append] always creates a fresh {!t}. *)
```

Comment:
this used to be the wording in the original documentation, I introduced a paragraph on top of the file to make this slightly more clear. feel free to improve it in a subsequent PR

Reply:
Right, but what does `The return [append] always creates a fresh {!t}.` tell the programmer reading it? To me this feels like saying `this returns a {!t}`.

Reply:
Since cstructs themselves are immutable, it doesn't really matter whether they're shared or not, so I've always interpreted the freshness as applying to the underlying buffer. In particular, it means that `append x empty` makes a copy of `x`.

Saying that `sub` returns a fresh `t` does seem pointless (although the comment clarifies that the buffer is shared).